src/python/pants/util/dirutil.py

```diff
@@ -100,18 +100,34 @@ def safe_mkdir_for_all(paths):
       created_dirs.add(dir_to_make)
 
 
-def safe_file_dump(filename, payload, binary_mode=True):
+def safe_file_dump(filename, payload, binary_mode=True, mode='w'):
```

Comment:
Hm I see now why you were just using mode=w, because binary mode was true. 

I think it would be better to change this functions signature to one of these two:
- just mode
- binary_mode and append, both booleans. Append defaults to false

This avoids the possibility of trying to pass mode=wb and having binary_mode=true

Reply:
I agree, but removing an existing parameter is out of scope for this PR I think. I can open a new ticket for that?

Reply:
Opened #6543 about it, and begin stitching these together in a more reasonable way here.